cmd/juju/controller/destroy.go

```diff
@@ -590,6 +590,21 @@ func (c *destroyCommandBase) getControllerEnviron(
 	return env, nil
 }
 
+func (c *destroyCommandBase) getControllerCloudSpecFromStore(
+	ctx *cmd.Context,
+	store jujuclient.ClientStore,
+	controllerName string,
+) (environscloudspec.CloudSpec, error) {
+	_, params, err := modelcmd.NewGetBootstrapConfigParamsFunc(
```

Comment:
This means that the command will only work when run from the exact same client that was used to bootstrap. And these client bootstrap params we are trying to get rid of. Can't we make an api call to the controller to get the cloud info off the controller model? This allows any juju client to run the kill command.